.github/workflows/checks.yml

```diff
@@ -0,0 +1,87 @@
+name: CI
+
+on:
+  push:
+    branches: [master]
+  pull_request:
+
+permissions:
+  # Permission for checking out code
+  contents: read
+
+jobs:
+  build:
+    name: Check & Build Provider
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-go@v5
+        with:
+          go-version-file: 'go.mod'
+      - run: chmod +x ./scripts/gofmtcheck.sh
+      - name: gofmt Check
+        run: ./scripts/gofmtcheck.sh
+      - run: go mod tidy
+      - run: go mod vendor
+      - name: Check vendor for changes
+        run: git diff --exit-code
+      - name: Build
+        run: go build -v
+
+  # diff:
+  #   name: Check Generated Code Difference
+  #   runs-on: ubuntu-latest
+  #   steps:
+  #     - uses: actions/checkout@v4
+  #     - uses: actions/setup-go@v5
+  #       with:
+  #         go-version-file: 'go.mod'
+  #     - name: Generate provider code
+  #       run: go generate
+  #     - name: Check generated code for diffs
+  #       run: git diff --exit-code
+
+  acceptance:
+    name: Acceptance Tests
+    needs: [build]
+    runs-on: ubuntu-latest
+    env:
+      ACI_USERNAME: 'ansible_github_ci'
+      ACI_PASSWORD: 'sJ94G92#8dq2hx*K4qh'
+    concurrency:
+        group: tf-aci-ci-test-${{ matrix.apic_host.name }}
+        cancel-in-progress: false
+    strategy:
+      fail-fast: false
+      matrix:
+        apic_host:
+          # - name: v4.2

```

Comment:
We currently don't have a way to skip tests that can't run on older APIC versions.